### PR TITLE
docs: reword adr-013 and describe `SubtreeRootThreshold`

### PR DESCRIPTION
**Pull request by rootulp**
Closes https://github.com/celestiaorg/celestia-app/issues/1877
Closes https://github.com/celestiaorg/celestia-app/issues/1607

## Description
1. Rephrased a couple of sections for clarity. If reviewers don't think it's any clearer, I can revert.
1. Added a section on the threshold. Replaced usage of `MaxSquareSize` with `SubtreeRootThreshold` where applicable.
1. Update links. I don't see a good way to relative link to markdown files in this repo from Go code so I removed the obsolete Github permalinks. Open to ideas if reviewers have suggestions on what to do with the links in Godoc.

### CODE REVIEW

**Comment by cmwaters** [comment]
> Renamed file and title to "Non-Interactive Default Rules for Reduced Padding 2"

Why does it need to be renamed to two? Can't we just document the changes in the changelog

**Comment by rootulp** [comment]
> Why does it need to be renamed to two?

It doesn't need to be renamed but I renamed b/c I thought it was misleading that the ADR was titled "zero padding". With this proposal, there will still be padding but there is just less padding than ADR-009.

**Review by evan-forbes — DISMISSED** [review]
due to the rename, its difficult to spot the rephrases, but its probably fine

I'm not blocking on renaming, but perhaps we should keep that to a separate PR?

File: docs/architecture/adr-020-deterministic-square-construction.md [hunk 1]
```diff
@@ -80,6 +80,6 @@ The new algorithm will no longer need to check that the blobs are ordered by nam
 
 ## References
 
-- [ADR013](./adr-013-non-interactive-default-rules-for-zero-padding.md)
+- [ADR-013](./adr-013-non-interactive-default-rules-for-reduced-padding-2.md)
```
[the remark this answers is not in the document]
Comment (evan-forbes) [reply docs/architecture/adr-020-deterministic-square-construction.md:83]: perhaps we could come up with a different name? or perhaps @nashqueue has a more accurate title?

**Comment by cmwaters** [comment]
> It doesn't need to be renamed but I renamed b/c I thought it was misleading that the ADR was titled "zero padding". With this proposal, there will still be padding but there is just less padding than ADR-009.

Theoretically if you have a ridiculously large SubtreeRootThreshold there will be zero padding no matter how large the blob is. 

I actually don't mind renaming it to "reduced-padding" but I don't think the `-2` suffix adds anything (just my personal view)

File: docs/architecture/adr-020-deterministic-square-construction.md (hunk 1 above)
Comment (cmwaters) [inline docs/architecture/adr-020-deterministic-square-construction.md:83]:
```suggestion
- [ADR013](./adr-013-non-interactive-default-rules-for-reduced-padding-2.md)
```
What was the reasoning for changing the format here. It seem elsewhere we don't include the `-`

[the remark this answers is not in the document]
Comment (rootulp) [reply docs/architecture/adr-020-deterministic-square-construction.md:83]: Will revert to original name

Comment (rootulp) [reply docs/architecture/adr-020-deterministic-square-construction.md:83]: I was attempting to be consistent with how ADRs are referenced and always use the hyphen. Will revert the name change and link changes.

File: docs/architecture/adr-013-non-interactive-default-rules-for-zero-padding.md [hunk 2]
```diff
@@ -2,57 +2,69 @@
...
+This means small blobs are inefficient because they generate more potential padding for the data they provide. This is not ideal as we want to minimize padding. Since users do not pay for namespace padding shares, they may not be sufficiently incentivized to submit large blobs.
 
-In the naive approach if you align the blob at an index of one you get zero padding as each blob can follow the next. This would make the hash of each share a subtree root. In a square with N shares, you would always get N subtree roots. But having a blob inclusion proof of size N for large blobs is too much and unfeasible.
+In the naive approach if the block proposer aligned blobs one after another then there would be zero padding between blobs. Although this approach minimizes padding, it comes at the cost of large blob inclusion proofs because the hash of each share would be a subtree root of the proof. In a square with N shares, there would be N subtree roots. As a result, light nodes would need to download N subtree roots to verify blob inclusion proofs for this square which may be too large for resource constrained devices.
```
Comment (nashqueue) [inline docs/architecture/adr-013-non-interactive-default-rules-for-zero-padding.md:23]: In the last sentence, it is necessary to say that the blob inclusion proof for **a blob of size N** would include N subtree roots. Light Nodes don't have to download N subtree roots if they only need the inclusion of a small blob.

File: docs/architecture/adr-013-non-interactive-default-rules-for-zero-padding.md [hunk 3]
```diff
@@ -2,57 +2,69 @@
...
 
-Let's assume a good threshold assumption is that the number of subtree roots in a blob inclusion proof is acceptable if it is smaller than the `MaxSquareSize`. This would mean that the blobs smaller than the square size can use the index of one to get zero padding. Blobs that are larger than `MaxSquareSize` but smaller than `MaxSquareSize * 2` can use the index of two to get a maximum of 1 padding square. Blobs that are larger than `MaxSquareSize * 2` but smaller than `MaxSquareSize * 4` can use the index of 4 to get a maximum of 3 padding shares and so on.
+Given this threshold, the new non-interactive default rules would be:
 
-The new non-interactive default rules would be:
+- If the blob length (in number of shares) is smaller than `SubtreeRootThreshold` then the blob starts at an index that is a multiple of 1.
+- If the blob length (in number of shares) is larger than `SubtreeRootThreshold` then the blob starts at an index that is a multiple of the blob length divided by `SubtreeRootThreshold` rounded up.
```
Comment (nashqueue) [inline docs/architecture/adr-013-non-interactive-default-rules-for-zero-padding.md:32]: Should we add the fact that we always take min of ADR9 and ADR13 ?

File: docs/architecture/adr-013-non-interactive-default-rules-for-zero-padding.md [hunk 4]
```diff
@@ -68,9 +80,9 @@ Here is a diagram of the worst-case padding for a threshold of 16 for the square
 
 ### Additional Remarks
 
-If the threshold is bigger than `MinSquareSize` for a particular blob then the blob will be aligned to the index of the `MinSquareSize` of the blob. This would prevent some blob size ranges to have higher padding than they had before this change. So the real new non-interactive default rules would be:
+If the threshold is bigger than `MinSquareSize(blob)` then the blob will be aligned to the index of the `MinSquareSize(blob)`. This would prevent some blob size ranges to have higher padding than they had before this change. So the real new non-interactive default rules would be:
 
-Blobs start at an index that is equal to a multiple of the blob length divided by `MaxSquareSize` rounded up. If this index is larger than the `MinSquareSize` of the blob then the blob starts at the index of the `MinSquareSize`.
+Blobs start at an index that is equal to a multiple of the blob length divided by `MaxSquareSize` rounded up. If this index is larger than the `MinSquareSize(blob)` then the blob starts at the index that is a multiple of of the `MinSquareSize(blob)`.
```
Comment (nashqueue) [inline docs/architecture/adr-013-non-interactive-default-rules-for-zero-padding.md:85]: Maybe we reference that where we explain the new non-interactive default rules? See my comment above

**Review by nashqueue — COMMENTED** [review]
Thank you! Left some comments

File: docs/architecture/adr-013-non-interactive-default-rules-for-zero-padding.md (hunk 3 above)
Comment (rootulp) [reply docs/architecture/adr-013-non-interactive-default-rules-for-zero-padding.md]: Resolved by https://github.com/celestiaorg/celestia-app/pull/1959/commits/4a4d33628a8d8c6f1f7664f34d18d7c4e9d77830

File: docs/architecture/adr-013-non-interactive-default-rules-for-zero-padding.md (hunk 2 above)
Comment (rootulp) [reply docs/architecture/adr-013-non-interactive-default-rules-for-zero-padding.md]: Resolved by https://github.com/celestiaorg/celestia-app/pull/1959/commits/4a4d33628a8d8c6f1f7664f34d18d7c4e9d77830

**Review by evan-forbes — APPROVED** [review]

**Review by cmwaters — APPROVED** [review]

File: docs/architecture/adr-013-non-interactive-default-rules-for-zero-padding.md (hunk 4 above)
Comment (nashqueue) [reply docs/architecture/adr-013-non-interactive-default-rules-for-zero-padding.md]: It is good. You are introducing subtree root width as a variable name, but explaining what it means in words may also make sense as it is the first mention in the ADR.

Comment (rootulp) [reply docs/architecture/adr-013-non-interactive-default-rules-for-zero-padding.md]: Was [this footnote](https://github.com/celestiaorg/celestia-app/pull/1959/files#diff-d55bfa04f022c5fcffa40f2fe89ff8f7047a6367f2f0ba73eab3e40c3923d5eaR105) not easy to discover? or are there other words you would use to define it?

Comment (rootulp) [reply docs/architecture/adr-013-non-interactive-default-rules-for-zero-padding.md]: I'm going to merge b/c two approvals but happy to amend the footnote or describe it differently in a follow-up PR. Thanks for your reviews!

**Pull request merged**